File: frontend/moviepumpkins-web/features/media.reviews/model/Review.ts
export interface Review {
    id: number;
    user: {
        username: string;
        displayName: string;
    };
    title: string;
    content: string;
    spoilerFree: boolean;
    createdAt: string;
    modifiedAt: string;
    likes: number;
    dislikes: number;
    userOwnRating?: "LIKE" | "DISLIKE" | "NO_RATING" | undefined;
}

export interface Reviews {
    pageCount: number;
    currentPageReviews: Review[];
    page: number;
}

File: backend-core/src/main/kotlin/net/moviepumpkins/core/media/review/model/Review.kt
package net.moviepumpkins.core.media.review.model

import net.moviepumpkins.core.app.model.UserAccount
import java.time.OffsetDateTime

open class ReviewContent(
    open val title: String,
    open val content: String,
    open val spoilerFree: Boolean,
)

data class Review(
    val id: Long,
    override val title: String,
    override val content: String,
    override val spoilerFree: Boolean,
    val likes: Int,
    val dislikes: Int,
    val createdAt: OffsetDateTime,
    val modifiedAt: OffsetDateTime,
    val creator: UserAccount,
    val userOwnRating: ReviewRatingType? = null,
) : ReviewContent(title, content, spoilerFree)